src/LuigisBoxBundle/ValueObject/SearchUrlBuilder.php

```diff
@@ -0,0 +1,319 @@
+    }
+
+    public function enableQueryUnderstanding(): self
+    {
+        if (null === $this->userId) {
+            throw new \InvalidArgumentException('Set $userId first.');
+        }
+
+        $this->qu = true;
+
+        return $this;
+    }
+
+    public function addFilter(string $key, string $value): self
+    {
+        $this->filters[$key] = $this->filters[$key] ?? [];
+        if (\is_string($this->filters[$key])) {
+            $this->filters[$key] = [$this->filters[$key]];
+        }
+        $this->filters[$key][] = $value;
+
+        $this->filters[$key] = array_unique($this->filters[$key]);
+
+        return $this;
+    }
+
+    public function setFilters(array $filters): self
+    {
+        $keys = array_keys($filters);
+        Assert::allString($keys, 'All filters keys must be string.');
+
+        $this->filters = $filters;
+
+        return $this;
+    }
+
+    public function setSize(int $size): self
+    {
+        $this->size = $size;
+
+        return $this;
+    }
+
+    public function setSort(string $sort, string $direction): self
+    {
+        Assert::oneOf(
+            $direction,
+            self::AVAILABLE_ORDER_DIRECTIONS,
+            sprintf('The sort direction parameter must by one of %s.', implode(',', self::AVAILABLE_ORDER_DIRECTIONS))
+        );
+
+        $this->sort = $sort . self::ARRAY_ITEM_SEPARATOR . $direction;
+
+        return $this;
+    }
+
+    public function setQuicksearchTypes(array $quicksearchTypes): self
+    {
+        Assert::allString($quicksearchTypes);
+
+        $this->quicksearchTypes = implode(self::LIST_SEPARATOR, $quicksearchTypes);
+
+        return $this;
+    }
+
+    public function setFacets(array $facets): self
+    {
+        $this->facets = implode(self::LIST_SEPARATOR, $facets);
+
+        return $this;
+    }
+
+    public function setFixits(bool $fixits): self
+    {
+        $this->useFixits = $fixits;
+
+        return $this;
+    }
+
+    public function addPrefer(string $key, string $value): self
+    {
+        $this->prefer[$key] = $this->prefer[$key] ?? [];
+        if (\is_string($this->prefer[$key])) {
+            $this->prefer[$key] = [$this->prefer[$key]];
+        }
+        $this->prefer[$key][] = $value;
+
+        $this->prefer[$key] = array_unique($this->prefer[$key]);
+
+        return $this;
+    }
+
+    public function setPreferArray(array $preferArray): self
+    {
+        $this->prefer = $preferArray;
+
+        return $this;
+    }
+
+    public function setHitFields(array $fields): self
+    {
+        Assert::allString($fields);
+
+        $this->hitFields = implode(self::LIST_SEPARATOR, $fields);
+
+        return $this;
+    }
+
+    public function setContext(Search\Context $context): self
+    {
+        $this->context = $context;
+
+        return $this;
+    }
+
+    public function setUserId(string $userId): self
+    {
+        $this->userId = $userId;
+
+        return $this;
+    }
+
+    public function toUrl(): string
```

Comment:
`toQuery` or `buildQuery` would be more suitable I think, the method is returning just that, not the full URL

Reply:
I have here query field, so changed it to `toUrlQuery()`